File: BankAccount/BankAccount.py

class BankAccount:
    # don't forget to add some default values for these parameters!
    

    def __init__(self, int_rate, balance): 
        self.int_rate = int_rate
        self.balance = balance
    def deposit(self, amount):
        self.balance= self.balance+amount
    def withdraw(self, amount):
        self.balance = self.balance-amount
    def display_account_info(self):
        balance = self.balance
        print(balance)
    def yield_interest(self):
        if self.balance>0:
            x= self.balance * self.int_rate
            self.balance = self.balance + x
    
bank1 = BankAccount(0.05,100)
bank2 = BankAccount(0.05,200)
bank1.deposit(300)
bank1.deposit(200)
bank1.deposit(600)
bank1.withdraw(100)
bank1.yield_interest()
bank1.display_account_info()

bank2.deposit(600)
bank2.deposit(500)
bank2.withdraw(100)
bank2.withdraw(400)
bank2.withdraw(10)
bank2.withdraw(1)
bank2.yield_interest()
bank2.display_account_info()

File: BankAccount/BankAccount.java
public class BankAccount{
    private double checkingBalance;
    private double savingsBalance;
    public static int numAccounts;
    public static int totalMoneyStored;

    public BankAccount(){
        this.checkingBalance = 0;
        this.savingsBalance = 0;
        numAccounts++;
    }

    public double getCheckingBalance(){
        return this.checkingBalance;
    }

    public double getSavingBalance(){
        return this.savingsBalance;
    }

    public void deposit(double amount, String type) {
        if (type == "checking") {
            this.checkingBalance += amount;
            totalMoneyStored += amount;
            System.out.println("$" + amount + " was deposited into your checkings account");
            }
        else if (type == "savings") {
            this.savingsBalance += amount;
            totalMoneyStored += amount;
            System.out.println("$" + amount + " was deposited into your savings account ");
        }
    }
    

    public void withdraw(int amount,String type){
        if (type == "checking") {
            if (amount <= this.checkingBalance) {
                this.checkingBalance -= amount;
                totalMoneyStored -= amount;
                System.out.println("$" + amount + " was withdrawn from your checking account");
            }
            else {
                System.out.println("insufficient funds");
            }
        }
        else if (type == "savings") {
            if (amount <= this.savingsBalance) {
                this.savingsBalance -= amount;
                totalMoneyStored -= amount;
                System.out.println("$" + amount + " was withdrawn from your saving account");
            }
            else {
                System.out.println("insufficient funds");
            }
    }
    }

    public String getBalances(){
        String str = "Savings Balance: $"+ this.savingsBalance+" | Checking Balance: $"+ this.checkingBalance;
        return str;
    }
}